dm/docs/RFCS/20220721_continuous_data_validation.md

```diff
@@ -0,0 +1,64 @@
+# Continuous Data Validation
+
+## Background
+
+TiDB already has [`sync-diff-inspector`](https://docs.pingcap.com/tidb/stable/sync-diff-inspector-overview) to do data validation for full data migration, but for incremental data migration there's no such option. We can do full data validation using `sync-diff-inspector` during database downtime for maintance to validate all incremental data migrated since last full data validation, but its time window is too short for `sync-diff-inspector` to completed and it might be too late to find any incorrect-migrated rows. We need an option to do incremental data validation which can validate incremental data in a more real-time way so we can find incorrect-migrated rows eailer and has less pressure on upstream and downstream database so business queries can run normally.
+
+## Limition
+
+Some known limitions to this design:
+- table which needs validation should have primary key or non-null unique key
+- there should only be compatible DDL operations in binlog stream between current syncer location and validation location, to make sure validation to use current DDL in schema-tracker to validate row change in history binlog:
+  - there is no operation to change primary key or non-null unique key
+  - there is no column order change operation
+  - there is no drop column operation
+  - table in downstream which needs validation cannot be dropped
+- do not support validation for tasks which enable extend-column
+- do not support validation for tasks which enable event filtering by expression
+- TiDB implements floating point data types differently with MySQL, we take it as equal if its absolute difference < 10^-6
+- do not support validation for JSON and binary data types
+
+## Detailed Design
+
+### Life cycle of validator
+
+Validator can be enabled together with the task or enabled on the fly, the task which enables validator should has a incremental migration unit, i.e. syncer. Validator will be deleted when the task is deleted. Enabled validator will be in `running` state, it can be stopped manually or stopped when met error that cannot be recovered automatically, then it's in `stopped` state. Validator in `stopped` state can be started again to return to `running` state.
```

Comment:
```suggestion
Validator can be enabled together with the task or enabled on the fly, the task which enables validator should has an incremental migration unit, i.e. syncer. Validator will be deleted when the task is deleted. Enabled validator will be in `running` state, it can be stopped manually or stopped when meeting error that cannot be recovered automatically, subsequently turning to `stopped` state. Validator in `stopped` state can be started again and return to `running` state.
```